packages/polaris-viz/src/components/DonutChart/Chart.tsx

```diff
@@ -88,10 +90,11 @@ export function Chart({
       : 'horizontal';
 
   const longestLegendWidth = data.reduce((previous, current) => {
-    const estimatedLegendWidth = estimateLegendItemWidth(
-      current.name ?? '',
-      characterWidths,
-    );
+    const text = showLegendValues
+      ? `${current.name} ${current.data[0]?.value}` ?? ''
+      : current.name ?? '';
+
+    const estimatedLegendWidth = estimateLegendItemWidth(text, characterWidths);
```

Comment:
When showing the legend values we have to take into account the length of the values when determining the width of the legend